extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/logs/AnyValueBytes.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.incubator.logs;
+
+import io.opentelemetry.api.internal.OtelEncodingUtils;
+import java.util.Arrays;
+import java.util.Objects;
+
+final class AnyValueBytes implements AnyValue<byte[]> {
+
+  private final byte[] value;
+
+  private AnyValueBytes(byte[] value) {
+    this.value = value;
+  }
+
+  static AnyValue<byte[]> create(byte[] value) {
+    Objects.requireNonNull(value, "value must not be null");
+    return new AnyValueBytes(Arrays.copyOf(value, value.length));
+  }
+
+  @Override
+  public AnyValueType getType() {
+    return AnyValueType.BYTES;
+  }
+
+  @Override
+  public byte[] getValue() {
+    return value;
```

Comment:
Its a good question. Maybe ByteBuffer is a good answer here. I want to avoid unnecessary allocations. But TBH, all this AnyValue stuff is not great from an allocations standpoint, since we take any input and minimally wrap it another object which does nothing more than describe its type. I'm chalking that up as essential complexity, but still trying to be somewhat conscious of performance where possible.

Reply:
Figured out `ByteBuffer` has `asReadOnlyBuffer` method which creates a read only view of the buffer. The ergonomics are a bit bad because read only ByteBuffer throws on various methods like `ByteBuffer#array()`, but otherwise is exactly what we want. I think we go with ByteBuffer, and document in `AnyValue` that its a read only view.